src/catalog/database_catalog.cpp

```diff
@@ -1324,6 +1324,192 @@ std::vector<std::pair<common::ManagedPointer<storage::index::Index>, const Index
+                                     const std::string &name) {
+  // First, insert into pg_class
+  auto *const class_insert_redo = txn->StageWrite(db_oid_, postgres::CLASS_TABLE_OID, pg_class_all_cols_pri_);
+  auto *const class_insert_pr = class_insert_redo->Delta();
+
+  // Write the sequence_oid into the PR
+  auto sequence_oid_offset = pg_class_all_cols_prm_[postgres::RELOID_COL_OID];
+  auto *sequence_oid_ptr = class_insert_pr->AccessForceNotNull(sequence_oid_offset);
+  *(reinterpret_cast<sequence_oid_t *>(sequence_oid_ptr)) = sequence_oid;
+
+  const auto name_varlen = storage::StorageUtil::CreateVarlen(name);
+
+  // Write the name into the PR
+  const auto name_offset = pg_class_all_cols_prm_[postgres::RELNAME_COL_OID];
+  auto *const name_ptr = class_insert_pr->AccessForceNotNull(name_offset);
+  *(reinterpret_cast<storage::VarlenEntry *>(name_ptr)) = name_varlen;
+
+  // Write the ns_oid into the PR
+  const auto ns_offset = pg_class_all_cols_prm_[postgres::RELNAMESPACE_COL_OID];
+  auto *const ns_ptr = class_insert_pr->AccessForceNotNull(ns_offset);
+  *(reinterpret_cast<namespace_oid_t *>(ns_ptr)) = ns_oid;
+
+  // Write the kind into the PR
+  const auto kind_offset = pg_class_all_cols_prm_[postgres::RELKIND_COL_OID];
+  auto *const kind_ptr = class_insert_pr->AccessForceNotNull(kind_offset);
+  *(reinterpret_cast<postgres::ClassKind *>(kind_ptr)) = postgres::ClassKind::SEQUENCE;
+
+  // Set schema to NULL because sequences don't need schema
+  const auto index_schema_ptr_offset = pg_class_all_cols_prm_[postgres::REL_SCHEMA_COL_OID];
+  class_insert_pr->SetNull(index_schema_ptr_offset);
+
+  // Set next_col_oid to NULL because sequences don't need col_oid
+  const auto next_col_oid_offset = pg_class_all_cols_prm_[postgres::REL_NEXTCOLOID_COL_OID];
+  class_insert_pr->SetNull(next_col_oid_offset);
+
+  // Set sequence_ptr to NULL because it gets set by execution layer after instantiation
+  const auto index_ptr_offset = pg_class_all_cols_prm_[postgres::REL_PTR_COL_OID];
+  class_insert_pr->SetNull(index_ptr_offset);
+
+  // Insert into pg_class table
+  const auto class_tuple_slot = classes_->Insert(txn, class_insert_redo);
+
+  // Now we insert into indexes on pg_class
+  // Get PR initializers allocate a buffer from the largest one
+  const auto class_oid_index_init = classes_oid_index_->GetProjectedRowInitializer();
+  const auto class_name_index_init = classes_name_index_->GetProjectedRowInitializer();
+  const auto class_ns_index_init = classes_namespace_index_->GetProjectedRowInitializer();
+  TERRIER_ASSERT((class_name_index_init.ProjectedRowSize() >= class_oid_index_init.ProjectedRowSize()) &&
+                     (class_name_index_init.ProjectedRowSize() >= class_ns_index_init.ProjectedRowSize()),
+                 "Index buffer must be allocated based on the largest PR initializer");
+  auto *index_buffer = common::AllocationUtil::AllocateAligned(class_name_index_init.ProjectedRowSize());
```

Comment:
Optional: maybe consider not using a raw pointer directly.

Reply:
They are using the raw pointers correctly as it is allowed in this part of the code (catalog). This is not an issue.